crates/circuit/src/dag_circuit.rs

```diff
@@ -4596,6 +4626,51 @@ impl DAGCircuit {
         Ok(target_dag)
     }
 
+    /// Modify `self` to mark its qubits as physical.
+    ///
+    /// This deletes the information about the virtual registers, and replaces it with the single
+    /// (implicitly) physical register.  This method does not need to traverse the DAG, other than
+    /// to add any ancilla in/out nodes.
+    ///
+    /// The qubit indices all stay the same; effectively, this is the application of the "trivial"
+    /// layout.  If the incoming DAG is supposed to be considered physical, this method can be used
+    /// to ensure it is in the canonical physical form.
+    pub fn make_physical(&mut self, ancillas: usize) {
+        let virtuals = self.num_qubits() as u32;
+        // The strategy here is just to modify the qubit and quantum register objects entirely
+        // inplace; we maintain all relative indices, so we don't need to modify any interner keys.
+        let num_qubits = virtuals
+            .checked_add(
+                ancillas
+                    .try_into()
+                    .expect("number of ancillas should fit in a u32"),
```

Comment:
Is there a reason to not just use `u32` for ancillas? Still having the overflow check is probably good but the type mismatch seems a bit weird here.

Reply:
All the other `DAGCircuit` methods that take in a number of qubits took `usize` already (stemming from `with_capacity`, I believe), so I was just matching that.  The Python-space one takes `u32` on entry so PyO3 will just raise the `TypeError`.